src/Infrastructure/Repositories/UserRepository.cs

```diff
@@ -0,0 +1,48 @@
+using System.Data;
+using Application.Common.Interfaces;
+using Dapper;
+using Domain.Entities;
+
+namespace Infrastructure.Repositories;
+
+public class UserRepository : IUserRepository
+{
+    private readonly IDbConnection _connection;
+    private readonly IDbTransaction _transaction;
+
+    public UserRepository(IDbConnection connection, IDbTransaction transaction)
+    {
+        _connection = connection;
+        _transaction = transaction;
+    }
+    
+    public async Task<User> CreateUserAsync(User user)
+    {
+        var commandText = $"INSERT INTO " +
```

Comment:
name it sql is more than enough

Reply:
> name it sql is more than enough

Using `commandText` to differentiate between a "Command" and a "Query"

Reply:
This is at Data/Persistence level, the difference between a command and query should not be a matter here